js/ui/handler/touch_zoom_rotate.js

```diff
@@ -39,6 +39,13 @@ TouchZoomRotate.prototype = {
         this._startVec = p0.sub(p1);
         this._startScale = this._map.transform.scale;
         this._startBearing = this._map.transform.bearing;
+        this._scalingSignificantly = false;
+        this._rotatingSignificantly = false;
+        this._blockRotation = undefined;
```

Comment:
Yeah definitely agree with this - `this._gestureIntent` will make the code much clearer.